### PR TITLE
DEV-3918-auto-exec-bin

### PR DESCRIPTION
**Pull request by zamazzal**
…bin exercise

### CODE REVIEW

File: sh/tests/auto-exec-bin_test.sh [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+# Unofficial Bash Strict Mode
+set -euo pipefail
+IFS='
+'
+
+FILENAME="student/auto-exec-bin.sh"
+script_dirS=$(cd -P "$(dirname "$BASH_SOURCE")" &>/dev/null && pwd)
+
+setupbin() {
```
Comment (MSilva95) [inline sh/tests/auto-exec-bin_test.sh:11]: this is checking if auto-exec-bin.sh exists
If it exist the myBins folder will not be created right?

**Review by MSilva95 — CHANGES_REQUESTED** [review]
I get this as soon as i run the test:

```
./auto-exec-bin_test.sh 
bin already exists!
./auto-exec-bin_test.sh: line 26: bash: No such file or directory
```
and i have all the folders and solution in the correct place. im not sure what is happening...

File: sh/tests/auto-exec-bin_test.sh (hunk 1 above)
Comment (zamazzal) [reply sh/tests/auto-exec-bin_test.sh:11]: yes

Comment (zamazzal) [reply sh/tests/auto-exec-bin_test.sh:12]: yes i have a bug here, I have a uncorrected condition
fixed by checking for the binary

File: sh/tests/auto-exec-bin_test.sh [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        mkdir -p ~/myBins
+        curl -o ~/myBins/01exec https://assets.01-edu.org/devops-branch/01exec
+        chmod +x ~/myBins/01exec
+        echo "bin installed!"
+    fi
+}
+
+challenge() {
+    # run soultion script
+    export PATH=""
```
Comment (MSilva95) [inline sh/tests/auto-exec-bin_test.sh:25]: when i run the test i get this error:
"./auto-exec-bin_test.sh: line 28: 01exec: command not found"
The new path is never added as you can see in the print, that is my current path printed after the student script run, and it doesn't add the "~/myBins" to the path. so the test is stuck there. 
image

Comment (MSilva95) [reply sh/tests/auto-exec-bin_test.sh:25]: can you run the test in your computer without this error?

**Review by MSilva95 — CHANGES_REQUESTED** [review]
test is stuck because the new path is not added

File: sh/tests/auto-exec-bin_test.sh (hunk 2 above)
Comment (nprimo) [reply sh/tests/auto-exec-bin_test.sh:25]: I replaced the 01exec with a simple bash script - the one used before was not running on my machine (Ubuntu 22.04)

Comment (MSilva95) [reply sh/tests/auto-exec-bin_test.sh]: Nice!

**Review by MSilva95 — APPROVED** [review]
LGTM

**Pull request merged**